Author: Breno Leitao <breno.leitao@gmail.com>
Date:   Thu Mar 30 10:36:43 2017 -0300

    Freeradius fails to build on systems with LibreSSL.
    
    FreeRadius try to link against function X509_get0_extensions() that does
    not exist on LibreSSl, so the following "if" clausure does not work
    correct for LibreSSL, since OpenSSL has a defined macro
    OPENSSL_VERSION_NUMBER for LibreSSL bigger than
    0x10100000, and does not have X509_get0_extensions():
    
     #if OPENSSL_VERSION_NUMBER >= 0x10100000L
                    ext_list = X509_get0_extensions(client_cert);
    
    This patch guarantee that this check is not valid for LibreSSL, thus,
    using the fall back.

--- a/src/main/tls.c	2017-03-29 21:01:55.660935694 +0000
+++ b/src/main/tls.c	2017-03-29 21:02:28.900936152 +0000
@@ -2131,7 +2131,7 @@ int cbtls_verify(int ok, X509_STORE_CTX
 	}
 
 	if (lookup == 0) {
-#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L && !defined(LIBRESSL_VERSION_NUMBER)
 		ext_list = X509_get0_extensions(client_cert);
 #else
 		X509_CINF	*client_inf;
